modules/cloudflarepurger/src/Plugin/Purge/Purger/CloudFlarePurger.php

```diff
@@ -0,0 +1,210 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\cloudflare\Plugin\PurgePurger\CloudFlarePurger.
+ */
+
+namespace Drupal\cloudflarepurger\Plugin\Purge\Purger;
+use Drupal\Core\Cache\Cache;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\cloudflare\CloudFlareStateInterface;
+use Drupal\cloudflarepurger\EventSubscriber\CloudFlareCacheTagHeaderGenerator;
+use Drupal\purge\Plugin\Purge\Purger\PurgerBase;
+use Drupal\purge\Plugin\Purge\Purger\PurgerInterface;
+use Drupal\purge\Plugin\Purge\Invalidation\InvalidationInterface;
+use CloudFlarePhpSdk\ApiEndpoints\CloudFlareAPI;
+use CloudFlarePhpSdk\ApiEndpoints\ZoneApi;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Psr\Log\LoggerInterface;
+
+/**
+ * CloudFlare purger.
+ *
+ * @PurgePurger(
+ *   id = "cloudflare",
+ *   label = @Translation("CloudFlare"),
+ *   description = @Translation("Purger for CloudFlare."),
+ *   configform = "Drupal\cloudflare\Form\CloudFlareAdminSettingsForm",
+ *   types = {"tag", "url", "everything"},
+ *   multi_instance = FALSE,
+ * )
+ */
+class CloudFlarePurger extends PurgerBase implements PurgerInterface {
+
+  /**
+   * The settings configuration.
+   *
+   * @var \Drupal\Core\Config\Config
+   */
+  protected $config;
+
+  /**
+   * A logger instance.
+   *
+   * @var \Psr\Log\LoggerInterface
+   */
+  protected $logger;
+
+  /**
+   * Tracks rate limits associated with CloudFlare Api.
+   *
+   * @var \Drupal\cloudflare\CloudFlareStateInterface
+   */
+  protected $state;
+
+  /**
+   * ZoneApi object for interfacing with CloudFlare Php Sdk.
+   *
+   * @var \CloudFlarePhpSdk\ApiEndpoints\ZoneApi
+   */
+  protected $zoneApi;
+
+  /**
+   * The current cloudflare ZoneId.
```

Comment:
If it is a ID, why is it then just called 'zone'? Is 'zone' always the id? #nitpickingftw